intellij/src/main/java/services/kiotaVersionHandler.java

```diff
@@ -0,0 +1,144 @@
+package services;
+
+import com.thetransactioncompany.jsonrpc2.*;
+
+// The JSON-RPC 2.0 server framework package
+import com.thetransactioncompany.jsonrpc2.server.*;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class kiotaVersionHandler implements RequestHandler {
+
+    @Override
+    public String[] handledRequests() {
+        return new String[]{"getkiotaversion"};
+    }
+
+    @Override
+    public JSONRPC2Response process(JSONRPC2Request request, MessageContext requestCtx) {
+        if (request.getMethod().equals("getkiotaversion")) {
+            // Echo first parameter
+            //ArrayList<String> response = new ArrayList<String>();
+
+            String version = getkiotaversion();
+            return new JSONRPC2Response(version, request.getID());
+        } else {
+            // Method name not supported
+            return new JSONRPC2Response(JSONRPC2Error.METHOD_NOT_FOUND, request.getID());
+        }
+    }
+
+    public String getkiotaversion() {
+        String software = "kiota";
+
+        try {
+            Process process = new ProcessBuilder(software, "--version").start();
+
+            // Read the command output
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line;
+            StringBuilder output = new StringBuilder();
+
+            while ((line = reader.readLine()) != null) {
+                output.append(line).append("\n");
+            }
+
```

Comment:
I believe you're doing that work because you didn't start the RPC server, and you can't rely on the library to do the parsing work for you. Hopefully by switching to the RPC command instead, you'll be able to rely on your JSON RPC lib to parse the response for you.